File: src/main/java/com/donrobo/fpbg/blueprint/building/AssemblingMachine1.java
package com.donrobo.fpbg.blueprint.building;

import net.sf.json.JSONObject;

public class AssemblingMachine1 extends AbstractBuilding {

    private final String recipe;

    public AssemblingMachine1(String recipe, int x, int y) {
        super(x, y);
        this.recipe = recipe;
    }

    @Override
    public int getWidth() {
        return 3;
    }

    @Override
    public int getHeight() {
        return 3;
    }

    @Override
    public AssemblingMachine1 move(int x, int y) {
        return new AssemblingMachine1(recipe, getX() + x, getY() + y);
    }

    @Override
    public String getName() {
        return "assembling-machine-1";
    }

    public String getRecipe() {
        return recipe;
    }

    @Override
    protected void addCustomPropertiesToJson(JSONObject json) {
        json.put("recipe", recipe);
    }

    @Override
    protected double getBlueprintXOffset() {
        return 1;
    }

    @Override
    protected double getBlueprintYOffset() {
        return 1;
    }
}

File: src/main/kotlin/com/donrobo/fpbg/blueprint/building/AssemblingMachine1.kt
package com.donrobo.fpbg.blueprint.building

import com.donrobo.fpbg.data.Int2
import net.sf.json.JSONObject

class AssemblingMachine1(val recipe: String, x: Int, y: Int) : AbstractBuilding(x, y) {

    override val width: Int
        get() = 3

    override val height: Int
        get() = 3

    override fun move(x: Int, y: Int): AssemblingMachine1 {
        return AssemblingMachine1(recipe, this.x + x, this.y + y)
    }

    override val name: String
        get() = "assembling-machine-1"

    override val visualizationCharacter: Char
        get() = '1'

    override fun addCustomPropertiesToJson(json: JSONObject) {
        json.put("recipe", recipe)
    }

    override val blueprintXOffset: Double
        get() = 1.0
    override val blueprintYOffset: Double
        get() = 1.0

    override fun rotateCCW(around: Int2, count: Int): Building {
        TODO("Untested")
//        val newPosition = position.rotateCCW(around, count) //probably some magic needed here
//
//        return AssemblingMachine1(recipe, newPosition.x, newPosition.y)
    }
}
